src/instrumentation/page.ts

```diff
@@ -0,0 +1,85 @@
+import { ReadableSpan } from '@opentelemetry/sdk-trace-base'
+import { Initialiser, setConfig } from '../config'
+import { exportSpans, proxyExecutionContext } from './common'
+import { Exception, SpanKind, SpanOptions, SpanStatusCode, context as api_context, trace } from '@opentelemetry/api'
+import { wrap } from '../wrap'
+import {
+	gatherIncomingCfAttributes,
+	gatherRequestAttributes,
+	gatherResponseAttributes,
+	getParentContextFromRequest,
+} from './fetch'
+
+type PageHandlerArgs = Parameters<PagesFunction>
+
+let cold_start = true
+export function executePageHandler(pagesFn: PagesFunction, [request]: PageHandlerArgs): Promise<Response> {
+	const spanContext = getParentContextFromRequest(request.request)
+
+	const tracer = trace.getTracer('pagesHandler')
+	const attributes = {
+		['faas.trigger']: 'http',
+		['faas.coldstart']: cold_start,
+		['faas.invocation_id']: request.request.headers.get('cf-ray') ?? undefined,
+	}
+	cold_start = false
+	Object.assign(attributes, gatherRequestAttributes(request.request))
+	Object.assign(attributes, gatherIncomingCfAttributes(request.request))
+	const options: SpanOptions = {
+		attributes,
+		kind: SpanKind.SERVER,
+	}
+
+	const promise = tracer.startActiveSpan(
+		`${request.request.method} ${request.functionPath}`,
```

Comment:
We recently introduced a new convention for instrumented spans, could you prefix this and the two `updateNames` to be prefixed with `fetchHandler`? Again, we could call them `functionHandler`, but with convergence a consistent name between Workers and Pages will make more sense in my opinion.